### ISSUE TITLE
[Function Behavior] loglikelihood Function Returns Loss Instead of Log Likelihood for Multiple Choice Selection

### ISSUE THREAD

**Issue by j1u2l3i4a5n** [EvolvingLMMs-Lab/lmms-eval#370]
### Summary

The project defines two primary model functions: `loglikelihood`and `generate_until`. The `loglikelihood` function, however, returns the *loss* rather than a log-likelihood measure for the input text. In the `multiple_choice` setting, the choice with the highest log-likelihood is selected as the predicted answer.

### Problem Details

In multiple-choice tasks, the `loglikelihood` function is used to select an answer choice by choosing the one with the highest log likelihood. 
https://github.com/EvolvingLMMs-Lab/lmms-eval/blob/325e16c0f64b68cbafde44c5daa24ef044b3b128/lmms_eval/api/task.py#L1447

However, since this function actually returns the *loss*, selecting the option with the *highest loss* may be unintended. 
https://github.com/EvolvingLMMs-Lab/lmms-eval/blob/325e16c0f64b68cbafde44c5daa24ef044b3b128/lmms_eval/models/llava.py#L275

It seems the intended behavior should be to select the option with the *smallest loss*, reflecting the choice with the highest likelihood.

Thank you for looking into this!

**Comment by kcz358** [comment]
Yeah, I think this could be an issue. I haven't looked into this closely because I simply write this to calculate perplexity. Maybe I should change it to argmin since most of our model return loss from their forward function

**Comment by Luodian** [comment]
Closing due to inactivity (463 days). Feel free to reopen if still relevant!
